src/liballoc/heap.rs

```diff
@@ -16,6 +16,18 @@
 
 use core::{isize, usize};
 
+#[allow(improper_ctypes)]
+extern {
+    #[allocator]
```

Comment:
What is the difference in meaning when `#[allocator]` is applied to an extern block vs a crate?